sdk/storage/Azure.Storage.Blobs/src/Models/CopyFromUriOperation.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Specialized;
+
+// TODO: Make this type internal once Operation<T> has an Id property
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// An <see cref="Operation{Int64}"/> for tracking the status of a 
+    /// <see cref="BlobBaseClient.StartCopyFromUriAsync"/> request.  Its
+    /// <see cref="Value"/> upon succesful completion will be the number of
+    /// bytes copied.
+    /// </summary>
+    public class CopyFromUriOperation : Operation<long>
+    {
+        /// <summary>
+        /// The client used to check for completion.
+        /// </summary>
+        private readonly BlobBaseClient _client;
+
+        /// <summary>
+        /// The CancellationToken to use for all status checking.
+        /// </summary>
+        private readonly CancellationToken _cancellationToken;
+
+        /// <summary>
+        /// ID of the copy operation.
+        /// </summary>
+        private readonly string _copyId;
+
+        /// <summary>
+        /// Gets the ID of the copy operation that can be compared with the
+        /// <see cref="BlobProperties.CopyId"/> value returned from
+        /// <see cref="BlobBaseClient.GetPropertiesAsync" /> or passed to
+        /// <see cref="BlobBaseClient.AbortCopyFromUriAsync" />.
+        /// </summary>
+        public virtual string Id => this._copyId;
+
+        /// <summary>
+        /// Whether the operation has completed.
+        /// </summary>
+        private bool _hasCompleted;
+
+        /// <summary>
+        /// Gets a value indicating whether the operation has completed.
+        /// </summary>
+        public override bool HasCompleted => this._hasCompleted;
+
+        /// <summary>
+        /// Whether the operation completed succesfully.
+        /// </summary>
+        private bool _hasValue;
+
+        /// <summary>
+        /// Gets a value indicating whether the operation completed and
+        /// succesfully produced a value.  The <see cref="Value"/> property
+        /// is the number of bytes copied by the operation.
+        /// </summary>
+        public override bool HasValue => this._hasValue;
+
+        /// <summary>
+        /// Initializes a new <see cref="CopyFromUriOperation"/> instance for
+        /// mocking.
+        /// </summary>
+        internal CopyFromUriOperation(
+            string copyId,
+            bool hasCompleted,
+            long? value = default,
+            Response rawResponse = default)
+        {
+            this._copyId = copyId;
+            this._hasCompleted = hasCompleted;
+            if (value != null)
```

Comment:
why not just store the nullable directly????

Reply:
Not going to block on this, but this sort of thing raises big red maintainability flags for me

Reply:
What's the difference between `Value == null` while the status is Pending and `Value == null` when the status is Failed?  You need an extra indicator somewhere.

Reply:
And to add, `HasCompleted` was originally going to be `OperationStatus` or similar but was simplified to `HasCompleted` to be as general purpose as possible.